examples/server/server.cpp

```diff
@@ -1923,6 +1954,57 @@ struct server_context {
         return slot.has_next_token; // continue
     }
 
+    void populate_token_probs(const server_slot & slot, completion_token_output & result, bool post_sampling, bool special, int idx) {
+        size_t n_probs = slot.params.sampling.n_probs;
+        int n_vocab = llama_n_vocab(llama_get_model(ctx));
+        if (post_sampling) {
```

Comment:
Is this check inverted? Post sampling should mean after the samplers have been applied.

Reply:
Yes it's reverted. The test cannot detect this mistake because the test model (stories260K) only output either 1.0 or 0.0 probability. I need to add a specific test case for it:

```
assert any(prob["prob"] == 1.0 for prob in tok["top_probs"])
```